pkgs/development/python-modules/pywalfox/default.nix

```diff
@@ -0,0 +1,25 @@
+{ lib, buildPythonPackage, fetchPypi }:
+
+buildPythonPackage rec {
```

Comment:
my understanding is pywalfox is an application an not intended to be imported as python library

if that's the case, move this out of python-modules into pkgs/by-name

here is some more information https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md

also instead of using `buildPythonPackage`, use `buildPythonApplication`

Reply:
Yes, you are correct. To be fair I wasn't and I still am not sure about moving it out of the python module namespace due to the naming of the packages by the author. What I mean by that is that

- the python name for the native messenger is `pywalfox`
- which is also the official name for the firefox addon

And since the nixpkgs naming convention is to keep the pypi/python package name I thought that atleast keeping it in the python namespace makes it clear that **this is the native messaging host**.

However, one counterargument might be that the probability of Pywalfox (the ff/thun/ addon) being in nixpkgs is low since most of the people using it get it via home-manager and nur....

If you think it's the correct choice I'll move it ofc :+1:

Reply:
- only python packages that intended to be imported in python are placed in python-modules.
- nixpkgs is currently in process  of migrating from all-packages.nix into by-name structure, therefore all new packages ( except some things  like python modules ) go there.

as for name clash that you are concerned about, In nixpkgs we can choose any name for it that makes sense.

i this case more correct name for package would be `pywalfox-native` ? ( opened for suggestions )

Reply:
> A possible solution might be to name the package `pywalfox-native` maybe? But we can leave it at `pywalfox` as well I guess.

worth noting that nixpkgs packags kinda have two different places where packages are given name.

one os `pname`, package name attribute in derivation, this is what tools like nix-env show when you install pacakges

second name is name of attribute in nixpkgs that reefers to the package, eg nix-env -iA nixpkgs.packagename

I propose to change both to `pywalfox-native`

Reply:
Thanks! I agree! I cleaned all of the redundancy, renamed the commit & PR and separated the maintainers addition into a separate commit. Thanks a lot for the patience. Hopefully I'll be able to return it to the community, thanks :sweat_smile:

Reply:
@tsandrini can you swap order of commits, in current order `pywalfox-native: init at 2.7.4` references maintainer that doesn't exist

Reply:
@kirillrdy I apologize for the wrong order! Should be okay now. :pray: